tools/automatic_query_fixer/src/main/java/com/google/cloud/bigquery/utils/queryfixer/service/BigQueryService.java

```diff
@@ -59,4 +76,29 @@ public BigQueryException catchExceptionFromDryRun(String query) {
 
     return null;
   }
+
+  /**
+   * Fetch the names of all table from the BigQuery given the project and dataset. If the project
+   * and/or dataset do not exist or not visible to the provided service account, {@link
+   * com.google.cloud.bigquery.BigQueryException} will be thrown.
+   *
+   * <p>It is possible that a dataset contains a large amount of tables, so this method will only
+   * fetch maximum 1000 of them.
+   *
+   * @param projectId project id
+   * @param datasetId dataset id
+   * @return list of table names belonging to the given project and dataset
+   */
+  public List<String> listTableNames(String projectId, String datasetId) throws BigQueryException {
+    DatasetId projectDatasetId = DatasetId.of(projectId, datasetId);
+    Page<Table> tables =
+        bigQuery.listTables(projectDatasetId, BigQuery.TableListOption.pageSize(TableFetchSize));
+    return StreamSupport.stream(tables.iterateAll().spliterator(), /* parallel= */ false)
```

Comment:
what if you set the "parallel= true", can you add a comment why you decided to set it to false and when we have to set it to true?

Reply:
Comment is added